tests/io/hdf5_test.py

```diff
@@ -201,7 +201,7 @@ def test_data_array_dtype_scipp_container():
     a.coords['1d'] = sc.empty(dims=x.dims, shape=x.shape, dtype=sc.DType.DataArray)
     for i in range(4):
         a.coords['1d'].values[i] = sc.DataArray(float(i) * sc.units.m)
-    a.coords['dataset'] = sc.scalar(sc.Dataset(data={'a': array_1d, 'b': array_2d}))
+    a.coords['dataset'] = sc.scalar(sc.Dataset(data={'a': array_1d}))
```

Comment:
Are we simply going to fail loading legacy files? Or should we return a DataGroup instead?

Reply:
Good question! I'd rather fail because
1. returning a datagroup would be tricky to implement (fallback mechanism like scippnexus)
2. It's unlikely that downstream code still works.
3. Our hdf5 files are not meant for long term storage. So the few files that could cause problems for use can be fixed manually.

Reply:
I suppose loading individual items as data arrays from such files would be a workaround that users can employ?

Reply:
Could be done. But requires knowing the file structure.